internal/server/iigo.go

```diff
@@ -22,10 +22,13 @@ func (s *SOMASServer) runIIGOEndOfTurn() error {
 	clientMap := getNonDeadClients(s.gameState.ClientInfos, s.clientMap)
 	for clientID, v := range clientMap {
 		tax := v.GetTaxContribution()
-		s.gameState.CommonPool += tax
-		newGameState := s.gameState.GetClientGameStateCopy(clientID)
-		newGameState.ClientInfo.Resources -= tax
-		v.GameStateUpdate(newGameState)
+
+		err := s.takeResources(clientID, tax, "tax")
+		if err != nil {
+			s.gameState.CommonPool += tax
+		} else {
+			s.logf("Error getting tax from %v: %v", clientID, err)
```

Comment:
do we just log n ignore?

Reply:
That should be fine, we just assume the client has made an invalid move and leave it alone

Reply:
Yeah I think this is the right way too.